src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -618,6 +643,64 @@ public Void get(long timeout, TimeUnit unit) {
 			throw new UnsupportedOperationException();
 		}
 
+		// this as discard hook
+		@Override
+		public void accept(I i) {
+			parent.sourceCleanup.accept(i);
+			// propagates discard to the downstream
+			Operators.onDiscard(i, actual.currentContext());
+		}
+
+		// Context interface impl
+		@Override
+		@SuppressWarnings("unchecked")
+		public <T> T get(Object key) {
+			if (KEY_ON_DISCARD.equals(key)) {
```

Comment:
Can we use reference equality

Reply:
Not in that case. The reference will be different https://github.com/reactor/reactor-core/blob/master/reactor-core/src/main/java/reactor/core/publisher/Hooks.java#L602 but we need to return an acceptor in case of the same content

Reply:
yep

Reply:
@OlegDokuka you could remove the `KEY_ON_DISCARD` by ensuring `currentContext` doesn't return `this` but `Operators.enableOnDiscard(actual.currentContext(), this)`, which would simplify things IMHO. The class can continue implementing `Consumer<?>` as the discarding hook.

Reply:
This is object overhead. The goal is to avoid this overhead. I can use `enabledOnDiscard` if we agree to not propagate released buffers do downstream hooks

Reply:
but on the other hand you're effectively creating a new `Context` instance every time you invoke `put`, `hasKey`, `delete`, `size` or `stream`... might as well pay that price upfront, which would additionally protect from the case where a new `Context` API is introduced with default method and you miss the `KEY_ON_DISCARD` in it.

Reply:
> which would additionally protect from the case where a new Context API is introduced with default method and you miss the KEY_ON_DISCARD in it.

https://github.com/reactor/reactor-netty/pull/1172#discussion_r445528928
Here is the hack. 


> but on the other hand you're effectively creating a new Context instance every time you invoke put, hasKey, delete, 

The probability of one using context is significantly lower compares to normal usage of `MonoSend`. Unless the spring framework does it somewhere above like  

```
outbound.send(
	flux.doOnDiscard()
)
```

and not 

```
outbound.send(flux)
        .doOnDiscard()
```

Reply:
There is no direct `doOnDiscard` for NettyOutbound

`NettyOutbound extends Publisher<Void>`

Reply:
Then. It makes no sense to support `Operator.onDiscard(v, actual.currentContext()` since `actual.currenctContext()` will always return empty context. Is not that true?

Reply:
if that is true, then I can simplify my changes